metric_learn/rca.py

```diff
@@ -8,47 +8,68 @@
 subsets of points that are known to belong to the same class.
 
 'Learning distance functions using equivalence relations', ICML 2003
+'Learning a Mahalanobis metric from equivalence constraints', JMLR 2005
 """
 
 from __future__ import absolute_import
 import numpy as np
 from six.moves import xrange
+from sklearn import decomposition
+import warnings
 
 from .base_metric import BaseMetricLearner
 from .constraints import Constraints
 
 
+# mean center each chunklet separately
+def _chunk_mean_centering(data, chunks, num_chunks):
+  chunk_mask = chunks != -1
+  chunk_data = data[chunk_mask]
+  chunk_labels = chunks[chunk_mask]
+  for c in xrange(num_chunks):
+    mask = chunk_labels == c
+    chunk_data[mask] -= chunk_data[mask].mean(axis=0)
+
+  return chunk_mask, chunk_data
+
+
 class RCA(BaseMetricLearner):
   """Relevant Components Analysis (RCA)"""
-  def __init__(self, dim=None):
+  def __init__(self, dim=None, pca_comps=None):
     """Initialize the learner.
 
     Parameters
     ----------
     dim : int, optional
         embedding dimension (default: original dimension of data)
+    pca_comps : int, float, None or string
+        see the sklearn documentation
```

Comment:
We should just explain what this is, and then maybe give a reference to sklearn. Maybe something like:
```
pca_comps : int or float, optional
    Number of components to keep during PCA preprocessing.
    If None (default), does not perform PCA.
    If ``0 < pca_comps < 1``, it is used as the minimum explained variance ratio.
    See sklearn.decomposition.PCA for more details.
```